kms-message/src/kms_message/kms_kmip_request.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KMS_KMIP_REQUEST_H
+#define KMS_KMIP_REQUEST_H
+
+#include "kms_message_defines.h"
+#include "kms_request.h"
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* kms_request_register_secretdata_new creates a KMIP Register request with
+ * a 96 byte SecretData payload.
+ * - len must be 96.
+ * - Callers must check for an error by calling kms_request_get_error. */
+KMS_MSG_EXPORT (kms_request_t *)
+kms_kmip_request_register_secretdata_new (void *reserved,
+                                          uint8_t *data,
+                                          uint32_t len);
+
+/* kms_request_activate_new creates a KMIP Activate request with the
+ * provided unique identifer.
+ * - unique_identifier must be a NULL terminated string.
+ * - Callers must check for an error by calling kms_request_get_error. */
+KMS_MSG_EXPORT (kms_request_t *)
+kms_request_activate_new (void *reserved, char *unique_identifier);
+
+/* kms_kmip_request_get_new creates a KMIP Get request with the provided unique
+ * identifer.
+ * - unique_identifier must be a NULL terminated string.
+ * - Callers must check for an error by calling kms_request_get_error. */
+KMS_MSG_EXPORT (kms_request_t *)
+kms_kmip_request_get_new (void *reserved, char *unique_identifier);
```

Comment:
```suggestion
kms_kmip_request_get_new (void *reserved, const char *unique_identifier);
```
Parameter unique_identifier should be const char * as function does not modify argument.